File: 49_group_anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string,vector<int>> m;
        string s;
        int i;
        for(i=0;i<strs.size();i++)
        {
            s=strs[i];
            sort(s.begin(),s.end());
            m[s].push_back(i);
        }
        vector<vector<string>> ans(m.size());
        i=0;
        for(auto str:m)
        {
            for(auto x:str.second)
            {
                ans[i].push_back(strs[x]);
            }
            i++;
        }
        return ans;
    }
};

File: 49_group_anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        result={}
        for i in strs:
            A=sorted(str(i))
            if not result or (str(A) not in result.keys()):
                result[str(A)]=[str(i)]
            elif(str(A) in result.keys()):
                result[str(A)].append(str(i))
            else:
                continue
        result_1=[]
        for i in result.values():
            result_1.append(i)
        return(result_1)
        